src/components/AlphabetFilter/AlphabetFilter.tsx

```diff
@@ -0,0 +1,197 @@
+      { text: "Show All", value: "showAll" },
+    ];
+
+    const refCurrentLetter = useRef(currentLetter);
+    const [selectedLetter, setSelectedLetter] = useState<string>(currentLetter);
+
+    // If the parent passes down a new currentLetter, and an onClick
+    // function exists, then set the internal state – selectedLetter –
+    // to the new currentLetter and update the refCurrentLetter with that value.
+    React.useEffect(() => {
+      if (
+        onClick &&
+        currentLetter &&
+        currentLetter !== refCurrentLetter.current
+      ) {
+        setSelectedLetter(currentLetter);
+        refCurrentLetter.current = currentLetter;
+      }
+    }, [currentLetter, onClick]);
+
+    /**
+     * This function is only called when clicking on a link should
+     * not update the URL or refresh the page.
+     */
+    const handleOnClick = (
+      e: React.MouseEvent | React.KeyboardEvent,
+      clickedLetter: string
+    ) => {
+      e.preventDefault && e.preventDefault();
+      if (selectedLetter === clickedLetter) return;
+      setSelectedLetter(clickedLetter);
+      onClick && onClick(clickedLetter);
+    };
+
+    const getButtonElement = (item) => {
+      const isSelectedLetter =
+        selectedLetter === item.value && item.value !== "showAll";
+      const isShowAll = item.value === "showAll";
+      const isButtonDisabled =
+        isDisabled ||
+        (activeLetters && !activeLetters.includes(item.value) && !isShowAll);
+
+      const buttonStyles = isSelectedLetter
+        ? { ...styles, border: "1px solid gray" }
+        : isShowAll
+        ? {
+            ...styles,
+            fontWeight: "normal !important",
+            whiteSpace: "nowrap",
+            marginLeft: "var(--nypl-space-m)",
+          }
+        : styles;
+
+      return (
+        <Button
+          aria-current={isSelectedLetter ? "page" : null}
+          aria-label={
+            item.text === "Show All" ? item.text : "Page " + item.text
+          }
+          buttonType="link"
+          id={`filter-${item.value}`}
+          isDisabled={isButtonDisabled}
+          __css={buttonStyles}
+          onClick={(e: React.MouseEvent | React.KeyboardEvent) => {
+            handleOnClick(e, item.value);
+          }}
+        >
+          {item.text}
+        </Button>
+      );
+    };
+
+    const getFilterLetters = () => {
+      if (isLargerThanMobile) {
+        return filterButtons.map((item) => {
+          return getButtonElement(item);
+        });
+      }
+      // split letters into rows of 10 + Show All
+      const mobileRows = filterButtons.reduce((rows, key, index) => {
+        if (index !== filterButtons.length - 1) {
+          return (
+            (index % 10 === 0
+              ? rows.push([key])
+              : rows[rows.length - 1].push(key)) && rows
+          );
+        } else {
+          return rows.push([key]) && rows;
+        }
+      }, []);
+      return (
+        <VStack sx={styles.mobileStack}>
```

Comment:
Do you think that maybe a `ul` list may be better than just buttons for accessibility? Perhaps a list of lists.

Reply:
Also, this does depend on @bigfishdesign13 's answer to whether it should be a row of 10 max characters per row. Otherwise, it could theoretically be a single list of buttons that wrap around.